.github/workflows/python_tests.yml

```diff
@@ -0,0 +1,29 @@
+name: CI - Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install uv, enable cache
+        uses: astral-sh/setup-uv@v4
+        with:
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
+
+      - name: Install the project
+        run: uv sync --all-extras --dev
+
+      - name: Run tests
+        run: uv run pytest tests
```

Comment:
Instead of adding this file, you could just add the unittest command in `uv.yml`.
Also, your tests are not using pytest, so the command you want is `uv run python -m unittest`

Reply:
Yes, I forgot you could just run commands using uv run...